### DISCUSSION TITLE
page.get_text('words') giving extra text

### DISCUSSION THREAD

**Discussion by kvrameshreddy** [pymupdf/PyMuPDF#1682] [Looking for help] [answered]
Hi @JorjMcKie,

I have a pdf, when I extract text, the extracted text has words which is not there on the pdf,

Can you help me solve this, I am not sure what is happening with this pdf file.

```
pdf2=fitz.open(r"D:\logs\b.pdf")
#text_in_pdf1.append(page1.get_text("words"))
for i in range(len(pdf2)):
    page2 = pdf2[i]
    page2.clean_contents()
    new_text2= page2.get_text("words")
    new_text2_list = [x[4] for x in new_text2]
```
this is the file I am using.
[b.pdf](https://github.com/pymupdf/PyMuPDF/files/8521104/b.pdf)

![image](https://user-images.githubusercontent.com/49742709/164222433-85388656-7431-458c-b22e-9f6441ef4cc3.png)

the words in the marked region are not available in the pdf file
![image](https://user-images.githubusercontent.com/49742709/164222205-0bf81303-bd34-4c41-8c4f-d09eb3885d12.png)

**Comment by JorjMcKie** [comment, marked as answer]
This is coded like so in the PDF. For whatever weird reason. Maybe to take some invisible notes.
You would have seen the crazy coordinates if you had looked at the words' coordinates.
You can only heal this by specifying the page rectangle as the clip: `page.get_text("words", clip=page.rect)`.
Here is the PDF source code as a proof:
![grafik](https://user-images.githubusercontent.com/8290722/164278800-3a75eca0-afae-4b4d-8aad-dc68446221f1.png)

**Reply by kvrameshreddy** [reply]
Thanks, @JorjMcKie for explaining this.
